### ISSUE TITLE
Multiple calls to the fit function resulting in too few model evaluations

### ISSUE THREAD

**Issue by marcoslbueno** [automl/auto-sklearn#903]
## Describe the bug ##
When I make multiple, sequential calls of fit() in a single script, it appears that only the first call is properly working. From the 2nd call onward, the number of models evaluated is drastically reduced. This happens even when more time budget is given to fit(), in the example below I set to 60 seconds.

For testing purposes, I placed each call to fit() in a python script and ran them separately, then the issue no longer appears. I suspect the problem could be related to memory management.

## To Reproduce ##
```
import sklearn.model_selection
import sklearn.datasets
import sklearn.metrics
from sklearn import preprocessing
import autosklearn.classification

X, y = sklearn.datasets.fetch_openml(data_id=40981, return_X_y=True)
X = X[:,0:3]

X_train, X_test, y_train, y_test = \
     sklearn.model_selection.train_test_split(X, y, random_state=1)

data = sklearn.datasets.fetch_openml(data_id=40981, as_frame=True)
feat_type = ['Categorical' if x.name == 'category' else 'Numerical' for x in data['data'].dtypes]
feat_type = feat_type[0:3]

cls = autosklearn.classification.AutoSklearnClassifier(
    time_left_for_this_task=60,
)
cls.fit(X_train, y_train, feat_type=feat_type)

predictions = cls.predict(X_test)
print("Accuracy score", sklearn.metrics.accuracy_score(y_test.astype('int'), predictions))

print(cls.sprint_statistics())

# Other call to fit, different seed
cls = autosklearn.classification.AutoSklearnClassifier(
    time_left_for_this_task=60,
    seed=2,
)
cls.fit(X_train, y_train, feat_type=feat_type)

predictions = cls.predict(X_test)
print("Accuracy score", sklearn.metrics.accuracy_score(y_test.astype('int'), predictions))

print(cls.sprint_statistics())

# I would then repeat this for say 100 times

``` 

## Expected behavior ##
In my laptop the 1st call to sprint_statistics() shows:

  **Dataset name: 8fd814888eb235d3690b348530299d07
  Metric: accuracy
  Best validation score: 0.660819
  Number of target algorithm runs: 30
  Number of successful target algorithm runs: 30
  Number of crashed target algorithm runs: 0
  Number of target algorithms that exceeded the time limit: 0
  Number of target algorithms that exceeded the memory limit: 0**

I would expect a similar Number of successful target algorithm runs from the 2nd call to sprint_statistics().

## Actual behavior, stacktrace or logfile ##
Please describe the expected behavior here. If there is a stacktrace, please paste it here. If there is no stacktrace printed, please upload the logfile which was stored in the `tmp_folder`

The 2nd call of sprint_statistics() shows:

  **Dataset name: 8fd814888eb235d3690b348530299d07
  Metric: accuracy
  Best validation score: 0.643275
  Number of target algorithm runs: 2
  Number of successful target algorithm runs: 1
  Number of crashed target algorithm runs: 0
  Number of target algorithms that exceeded the time limit: 1
  Number of target algorithms that exceeded the memory limit: 0**

## Environment and installation: ##

* OS: macOS Catalina 10.15.5
* Is your installation in a virtual environment or conda environment? Conda
* Python version 3.7.7
* Auto-sklearn version 0.7.0

**Comment by mfeurer** [comment]
Hey @marcoslbueno thanks for posting this issue. I cannot reproduce this on my Ubuntu machine, the output is:
```
Accuracy score 0.5895953757225434
auto-sklearn results:
  Dataset name: 8fd814888eb235d3690b348530299d07
  Metric: accuracy
  Best validation score: 0.643275
  Number of target algorithm runs: 26
  Number of successful target algorithm runs: 24
  Number of crashed target algorithm runs: 0
  Number of target algorithms that exceeded the time limit: 2
  Number of target algorithms that exceeded the memory limit: 0

Accuracy score 0.630057803468208
auto-sklearn results:
  Dataset name: 8fd814888eb235d3690b348530299d07
  Metric: accuracy
  Best validation score: 0.643275
  Number of target algorithm runs: 20
  Number of successful target algorithm runs: 15
  Number of crashed target algorithm runs: 0
  Number of target algorithms that exceeded the time limit: 5
  Number of target algorithms that exceeded the memory limit: 0
```
As we don't support OSX and can't test for that, could you please try running the script on a Unix machine?

**Comment by marcoslbueno** [comment]
Thanks @mfeurer. I reinstalled auto-sklearn within a new virtual environment and it now works fine. It was apparently conflicting packages in my previous installation.
